### PR TITLE
Don't force _WIN32_WINNT values

### PR DESCRIPTION
**Pull request by robUx4**
Forcing `_WIN32_WINNT` when the SDK has a default value is not a good idea. Assuming _WIN32_WINNT can be set to a certain value also has bad consequences and it may force some API's to be called when the (old) system doesn't have that. That should make the mbedtls build unsuable in that case.

Each commit explains why the change is needed. In the end `_WIN32_WINNT` is not used and either fallback calls are used or compilation errors are emited explaining what is wrong.

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required - minor tidy-up to build process
- [x] **backport** not required - not a bugfix
- [x] **tests** not required

### CODE REVIEW

**Comment by tom-cosgrove-arm** [comment]
@robUx4 (I've fixed the typos) "Forcing `_WIN32_WINNT` when the SDK has a default value is not a good idea" - can you give an authoritative reference for this?

Looking at https://learn.microsoft.com/en-gb/windows/win32/winprog/using-the-windows-headers, Microsoft documents how to set `_WIN32_WINNT` to get optional features, and it pretty clearly implies that this is expected

**Comment by robUx4** [comment]
> "Forcing _WIN32_WINNT when the SDK has a default value is not a good idea" - can you give an authoritative reference for this?

The wording is not correct. It's both the SDK default (in Visual Studio you can pick which SDK you want to use) and the forced value by the user (in my case VLC contribs which force a certain value depending on our target).

A project should never set this value on behalf of the user. In this case it was forcing the availabilty of some API for systems that did not have the API (for example if you target Windows 2000). It was also hiding some API's that are available in Windows UWP only since Windows 10 (so if you force Vista they are not available).

**Comment by robUx4** [comment]
I updated the code to not use windef.h which seems to have issues. Including socket headers will also pick the environment/default `_WIN32_WINNT` (checked in Win7+ SDKs and mingw-w64).

**Review by daverodgman — CHANGES_REQUESTED** [review]
One minor change, otherwise LGTM.

**Comment by daverodgman** [comment]
> A project should never set this value on behalf of the user.

Agreed. AIUI, over-riding `_WIN32_WINNT` is about defining the target system you want to build for, which is a decision for the user, not for Mbed TLS.

File: library/x509_crt.c [hunk 1]
```diff
@@ -1602,6 +1603,10 @@ int mbedtls_x509_crt_parse_path(mbedtls_x509_crt *chain, const char *path)
 
 cleanup:
     FindClose(hFind);
+#else /* !_WIN32_WINNT_XP */
+    /* FindFirstFileW not available, return an error */
+    ret = MBEDTLS_ERR_X509_FILE_IO_ERROR;
```
[the remark this answers is not in the document]
Comment (robUx4) [reply library/x509_crt.c:1605]: I guess it's OK since the code should have not built or link before that anyway.

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Pull request merged**